packages/plugins/on-resolve/src/index.ts

```diff
@@ -39,6 +39,7 @@ export type OnResolve<PluginContext extends Record<string, any> = {}> = (
  */
 export function useOnResolve<PluginContext extends Record<string, any> = {}>(
   onResolve: OnResolve<PluginContext>,
+  excludeIntrospection = true,
```

Comment:
Can you use an option object instead ? If we want to add more options like this in the future, we don't want to force users to give every parameter in the right order.

@ardatan Do you think it is worth a breaking change to replace the `onResolve` parameter with the object option ? Or do you think it's better to place it as a second parameter ?

Reply:
Yea no worries I considered it myself @EmrysMyrddin, to me it makes sense to have onResolve as the first argument and options as a second as you must supply an onResolve as otherwise the plugin has no value what do you think?
e.g
```ts
export function useOnResolve<PluginContext extends Record<string, any> = {}>(onResolve: OnResolve, opts?: UseOnResolveOptions): Plugin<PluginContext> {
```
This would keep it backwards compatible, or would you rather I wrap onResolve into the options?

Reply:
Both are possible :-) I'm not sure which one is the best. I prefer having one object with everything in it, for consistency with other plugins API, but I'm not sure it is worth the breaking change.

Reply:
I don't think adding an additional parameter with some options object is a breaking change.

Reply:
@ardatan Yes, that's the non breaking solution. The question was is it worth to replace the existing parameter with an option object, which would be breaking.

An other solution is to support both passing a function and an object, and you check at runtime in which case we are.

This allows to be non breaking, but still having the same API than other plugins with a unique parameter being an option object.

Reply:
> is it worth to replace the existing parameter with an option object, which would be breaking

I don't think we need to do this now, or at all? I think the API of `useOnResolve` is quite good where you provide the onResolve function (which is the goal of the plugin) first and then options optionally second.

@darren-west I'd just change to [this approach](https://github.com/n1ru4l/envelop/pull/1982/files#r1348433831) instead and we're good to go!

Reply:
pushed